nessie/billRequests.py

```diff
@@ -1,58 +1,56 @@
-import json
 import requests
 
-from nessie.models.bill import Bill
+from .models.bill import Bill
+from .utils.exceptions import NessieApiError
 
 class BillRequest():
 
     # perhaps need better way of injecting key?
     def __init__(self, key):
         self.base_url = "http://api.reimaginebanking.com"
         self.key = key
-        pass
-    
-    def get_account_bills(self, account_id):
-        url = f'{self.base_url}/accounts/{account_id}/bills/?key={self.key}'
-        raise NotImplementedError
-        pass
-
 
     # Get the bill using the bill_id
     # will return
-    # TODO: better way of error handling
-    # TODO: better way of appending id
+    # TODA: better way of error handling
+    # TODA: better way of appending id
     def get_bill(self, bill_id):
         url = f'{self.base_url}/bills/{bill_id}?key={self.key}' 
         # "/customers/56c66be5a73e492741507273/bills?key="
         response = requests.get(url)
-        # if (response.status_code != 200):
-        #    error_handle(response)
+        if (response.status_code != 200):
+            raise NessieApiError(response)
         result = response.json()
-        # need to refactor better
-        try:
-            result['bill_id']=result['_id']
-            del result['_id']
-        except KeyError:
-            pass
-        return result
+        return Bill(result)
 
     def get_customer_bills(self, customer_id):
         url = f'{self.base_url}/customers/{customer_id}/bills?key={self.key}'
         response = requests.get(url)
-        #error_handle(response)
-        return response.json()
+        if (response.status_code != 200):
+            raise NessieApiError(response)
+        # returns an array of bills in json format
+        result = response.json()
+        # for each bill in json format convert it to a Bill object
+        # using map operator and convert iterator to list
+        return list(map(Bill, result))
 
     def get_account_bills(self, account_id):
         url = f'{self.base_url}/accounts/{account_id}/bills?key={self.key}'
         response = requests.get(url)
-        return response.json()
-
-    #
-    def _create_bill_json(self, account_id, bill):
+        if (response.status_code != 200):
+            raise NessieApiError(response)
+        result = response.json()
+        return list(map(Bill, result))
+    
+    # allows users to create bills by passing in json
+    def _create_bill_json(self, account_id, bill_json):
         url = f'{self.base_url}/accounts/{account_id}/bills?key={self.key}'
-        response = requests.post(url,json=bill) # change json to Bill class??
+        response = requests.post(url,json=bill_json)
```

Comment:
idk if any validation on our side is possible for bills but if so it would be a good idea

Reply:
ive got some validation in the transaction branch (yeah i know should probably be in this branch lol)